qa/tasks/cephfs/test_cephfs_shell.py

```diff
@@ -560,6 +560,76 @@ def test_du_with_no_args(self):
                         "expected_output -\n{}\ndu_output -\n{}\n".format(
                         expected_output, du_output)
 
+class TestQuota(TestCephFSShell):
+    dir_name = 'testdir'
+
+    def create_dir(self):
+        mount_output = self.mount_a.run_shell('mkdir ' + self.dir_name)
+        log.info("cephfs-shell mount output:\n{}".format(mount_output))
+
+    def validate(self, input_val):
+        quota_output = self.get_cephfs_shell_cmd_output('quota set --max_bytes '+input_val[0]+' --max_files '+input_val[1]+ ' '+ self.dir_name)
+        log.info("cephfs-shell quota set output:\n{}".format(quota_output))
+
+        quota_output = self.get_cephfs_shell_cmd_output('quota get '+ self.dir_name)
+        log.info("cephfs-shell quota get output:\n{}".format(quota_output))
+
+        quota_output = quota_output.split()
+        return quota_output[1], quota_output[3]
```

Comment:
````
2019-10-13T01:34:23.873 INFO:tasks.cephfs_test_runner:======================================================================
2019-10-13T01:34:23.874 INFO:tasks.cephfs_test_runner:ERROR: test_exceed_file_limit (tasks.cephfs.test_cephfs_shell.TestQuota)
2019-10-13T01:34:23.874 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2019-10-13T01:34:23.874 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2019-10-13T01:34:23.874 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20191012.025926/qa/tasks/cephfs/test_cephfs_shell.py", line 610, in test_exceed_file_limit
2019-10-13T01:34:23.874 INFO:tasks.cephfs_test_runner:    self.test_set()
2019-10-13T01:34:23.874 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20191012.025926/qa/tasks/cephfs/test_cephfs_shell.py", line 583, in test_set
2019-10-13T01:34:23.874 INFO:tasks.cephfs_test_runner:    self.assertTupleEqual(self.validate(set_values), set_values)
2019-10-13T01:34:23.875 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20191012.025926/qa/tasks/cephfs/test_cephfs_shell.py", line 578, in validate
2019-10-13T01:34:23.875 INFO:tasks.cephfs_test_runner:    return quota_output[1], quota_output[3]
2019-10-13T01:34:23.875 INFO:tasks.cephfs_test_runner:IndexError: list index out of range
````

From: /ceph/teuthology-archive/pdonnell-2019-10-12_07:19:34-fs-wip-pdonnell-testing-20191012.025926-distro-basic-smithi/4407538/teuthology.log

Reply:
Strange, I don't see this error locally. This error should not occur if quota is set.
```
2019-10-15 16:05:48,098.098 INFO:tasks.cephfs.test_cephfs_shell:cephfs-shell mount output:
<__main__.LocalRemoteProcess object at 0x7f69336ebad0>
2019-10-15 16:05:48,098.098 INFO:tasks.cephfs.test_cephfs_shell:Running command: cephfs-shell -c ./ceph.conf -- quota set --max_bytes 6 --max_files 2 testdir
2019-10-15 16:05:48,098.098 INFO:__main__:Running ['cephfs-shell', '-c', './ceph.conf', '--', 'quota set --max_bytes 6 --max_files 2 testdir']
2019-10-15 16:05:48,365.365 INFO:tasks.cephfs.test_cephfs_shell:cephfs-shell quota set output:
max_bytes set to 6
max_files set to 2
2019-10-15 16:05:48,365.365 INFO:tasks.cephfs.test_cephfs_shell:Running command: cephfs-shell -c ./ceph.conf -- quota get testdir
2019-10-15 16:05:48,365.365 INFO:__main__:Running ['cephfs-shell', '-c', './ceph.conf', '--', 'quota get testdir']
2019-10-15 16:05:48,576.576 INFO:tasks.cephfs.test_cephfs_shell:cephfs-shell quota get output:
max_bytes: 6
max_files: 2
2019-10-15 16:05:48,577.577 INFO:__main__:Running ['touch', '/tmp/tmpG65zey/mnt.0/testdir/file1']
2019-10-15 16:05:48,590.590 INFO:__main__:Running ['touch', '/tmp/tmpG65zey/mnt.0/testdir/file2']
touch: cannot touch '/tmp/tmpG65zey/mnt.0/testdir/file2': Disk quota exceeded
2019-10-15 16:05:48,601.601 INFO:__main__:Running ['./bin/ceph', 'log', 'Ended test tasks.cephfs.test_cephfs_shell.TestQuota.test_exceed_file_limit']
2019-10-15 16:05:49,254.254 INFO:__main__:Running ['stat', '--file-system', '--printf=%T\n', '--', '/tmp/tmpG65zey/mnt.0']
2019-10-15 16:05:49,277.277 INFO:tasks.cephfs.fuse_mount:ceph-fuse is mounted on /tmp/tmpG65zey/mnt.0
2019-10-15 16:05:49,277.277 INFO:tasks.cephfs.fuse_mount:Running fusermount -u on local...
2019-10-15 16:05:49,277.277 INFO:__main__:Running ['fusermount', '-u', '/tmp/tmpG65zey/mnt.0']
2019-10-15 16:05:49,292.292 INFO:__main__:Running ['stat', '--file-system', '--printf=%T\n', '--', '/tmp/tmpG65zey/mnt.0']
2019-10-15 16:05:49,303.303 INFO:__main__:kill 
2019-10-15 16:05:49,303.303 INFO:__main__:kill: killing pid 6274 (['./bin/ceph-fuse', '-f', '--name', 'client.0', '/tmp/tmpG65zey/mnt.0'])
2019-10-15T16:05:46.935+0530 7f4746c98f40 -1 WARNING: all dangerous and experimental features are enabled.
2019-10-15T16:05:46.945+0530 7f4746c98f40 -1 WARNING: all dangerous and experimental features are enabled.
2019-10-15T16:05:46.948+0530 7f4746c98f40 -1 WARNING: all dangerous and experimental features are enabled.
2019-10-15T16:05:46.950+0530 7f4746c98f40 -1 init, newargv = 0x55ea65a5d2e0 newargc=7
ceph-fuse[6274]: starting ceph client
ceph-fuse[6274]: starting fuse
ceph-fuse[6274]: fuse finished with error 0 and tester_r 0
2019-10-15 16:05:49,304.304 INFO:__main__:Running ['rm', '-rf', '/tmp/tmpG65zey/mnt.0']
2019-10-15 16:05:49,313.313 INFO:__main__:test_exceed_file_limit (tasks.cephfs.test_cephfs_shell.TestQuota) ... ok
```

Reply:
@batrick 
Tests on this PR ran locally (on Fedora 29) successfully for me. Running on senta machine wasn't successful because py3 compatible Cython modules aren't built for Ubuntu by default. I suppose this is the case for the teuth tests too since it ran on a Ubuntu machine.